src/hosted-form/iframe-content/hosted-input-initializer.ts

```diff
@@ -54,6 +59,22 @@ export default class HostedInputInitializer {
             .toPromise();
     }
 
+    firefoxAndSafariBugfix(): void {
+        if (document.activeElement === document.body) {
+            const textField = document.querySelectorAll('input')[0];
+            if (textField) {
+                const browserName = navigator.userAgent.toLowerCase();
+                if (browserName.indexOf('firefox') > -1) {
+                    textField.focus();
+                } else if (browserName.indexOf('safari') > -1) {
+                    setTimeout(() => {
+                        textField.focus();
+                    }, 150);
```

Comment:
Have tested `0`, `100`. It appears that `150` is an acceptable wait time for Safari to be visually ready.  There is no obvious waiting experience in actual use.

Reply:
Is there any harm in applying `focus` on all browsers so we don't have to do browser detection?

Reply:
No, I don't think it's harmful as `document.activeElement === document.body` will detect a Firefox and Safari specific behaviour. It's possible that we only need to keep browser detection on line 69.

Reply:
Interesting finding on the timeout 😄 

Just out of curiosity what does the `document.activeElement` is for chrome 🤔 

Is that by design we do browser detection in line #35 as well?

Reply:
For Chrome, `document.activeElement` is always a text input field during the payment step. 
In this commit, I was thinking about applying a bugfix to Safari and Firefox first, then triggering different actions, hence I did two browser detections.

Reply:
Another question I have is, why `150` specifically? Does `0` also work?

Reply:
`0` is not working. Safari still requires two tabs to jump between fields.
I read about `SAFARI_FOCUS_TIMEOUT` concept on the Braintree's hosted field [code](https://github.com/braintree/braintree-web/blob/75c1c2496daa529525d7b6bf852fc0f3264ab7e0/src/hosted-fields/external/hosted-fields.js#L1375). Not exactly the same case, but I think it tells me Safari needs some time to work.

Reply:
Hi @davidchin @animesh1987 , I have updated the code.

1.  Apply the bugfix on all browsers.
2. Use Braintree's [method](https://github.com/braintree/braintree-web/blob/df38a9b1031504472939a1e7315becd69b7fabed/src/hosted-fields/internal/components/base-input.js#L249) to fix Safari's focus issue. The limitation is that this workaround is not working any more on the latest iOS 14.5.

Here is an updated demo for Safari.

https://user-images.githubusercontent.com/88361607/158500548-ee6e9a61-a4e0-4fbf-b37a-e6a7e42ad6dd.mov